plugins/modules/terraform.py

```diff
@@ -710,7 +879,9 @@ def process_complex_args(vars):
     if computed_state == 'absent' and workspace != 'default' and purge_workspace is True:
         remove_workspace(terraform_binary, project_path, workspace)
 
-    module.exit_json(changed=plan_file_needs_application, state=computed_state, workspace=workspace, outputs=outputs,
+    diff = dict(before=old_state_file, after=final_state_file)
```

Comment:
Could you rename the three `*_state_file` variables to just be `*_state`? These aren't always state files, especially with plans, and we don't read *.tfstate directly anyway. I'd also suggest renaming functions like `get_state_file_content_from_plan` to `get_state_from_plan` for clarity.

Reply:
Renamed as suggested